common/network-types/src/frame.rs

```diff
@@ -0,0 +1,347 @@
+use crate::errors::NetworkTypeError;
+use crate::errors::NetworkTypeError::{IncompleteFrame, InvalidFrameId, ReassemblerClosed};
+use crossbeam_skiplist::SkipSet;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use futures::{Sink, Stream};
+use std::pin::Pin;
+use std::sync::atomic::{AtomicU16, Ordering};
+use std::sync::OnceLock;
+use std::task::{Context, Poll};
+
+fn segment(data: &[u8], mtu: u16, frame_id: u16) -> Vec<Segment> {
```

Comment:
Since this protocol is meant to be in-message should we indicate that in the type naming? As is now it looks like these are native protocol types.

Reply:
What do you suggest?